### PR TITLE
Step1

### LINKED ISSUE #32: Step one: Adding layouts from Zeplin for Build Trust view
https://app.zeplin.io/project/598b122ec551c21f559ad6ae/screen/598c7985a1f0c405e0cc92cc 

Here's the Zeplin version

### PR DESCRIPTION
**Pull request by buzjuka**
# Pull request template

What does the code do?
The code is for responsive design of step 1 "Build trust"

What else should you know? (tips needed to use the code later on)
The code incudes new mixin "gm-button-icon" that should be used in all cases when it should be rendered button with favicon or svg

resolve #32

### CODE REVIEW

File: app/assets/stylesheets/_mixins.scss [hunk 1]
```diff
@@ -7,17 +7,70 @@
...
   box-shadow: 0 2px 4px 0 $shadow-5;
+  display: flex;
+  align-items: center;
 }
 @mixin flex-center {
   display: flex;
   align-items: center;
   justify-content: center;
 }
+@mixin gm-title {
```
Comment (k41n) [inline app/assets/stylesheets/_mixins.scss]: Use `extend-only selectors` for things like that

Comment (k41n) [inline app/assets/stylesheets/_mixins.scss]: http://sass-lang.com/documentation/file.SASS_REFERENCE.html#_extend-Only_Selectors__placeholders

File: app/assets/stylesheets/_mixins.scss [hunk 2]
```diff
@@ -7,17 +7,70 @@
...
+  }
+  &#{$divider}text {
+    display: inline-block;
+    @include btn-with-text($cerulean-2);
+  }
+  $scope: &;
+
+  &:hover {
+    #{$scope}#{$divider}icon {
+      background-color: #e6e6e6;
```
Comment (k41n) [inline app/assets/stylesheets/_mixins.scss]: Never specify color directly, use variables

File: app/assets/stylesheets/_mixins.scss [hunk 3]
```diff
@@ -7,17 +7,70 @@
...
+      border-color: #adadad;
+      color: $night-rider;
+      path {
+        fill: $night-rider;
+      }
+    }
+    #{$scope}#{$divider}text {
+      background-color: #e6e6e6;
```
Comment (k41n) [inline app/assets/stylesheets/_mixins.scss]: So you won't have to repeat

File: app/assets/stylesheets/social_media_forms/new.scss [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    display: flex;
+    flex-direction: column;
+    min-width: 18.4%;
+    align-items: center;
+    min-height: 12.5rem;
+    justify-content: center;
+    img {
+      margin-bottom: 1rem;
+    }
+    &_active {
```
Comment (k41n) [inline app/assets/stylesheets/social_media_forms/new.scss]: If we are speaking about BEM, let's markup modifiers with hyphen as specified in reference

File: app/assets/stylesheets/social_media_forms/new.scss [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+      }
+      &__button-check {
+        position: absolute;
+        margin: initial;
+        right: -2.5rem;
+        top: -3.5rem;
+      }
+    }
+  }
+}
```
Comment (k41n) [inline app/assets/stylesheets/social_media_forms/new.scss]: Please separate this file, 194 lines is too much and obviously there are components like button which are on its own and should not be encased this one

File: app/views/social_media_forms/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,68 @@
+<div class="gm-social-media-forms">
```
Comment (k41n) [inline app/views/social_media_forms/new.html.erb]: Why `erb`? Want my eyes melt?

File: app/views/social_media_forms/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  <div class="gm-social-media-forms__steps">
+    <div class="gm-social-media-forms__step gm-social-media-forms__step_active">
+      <%= image_tag 'social_media_forms/trust-score-active.svg' %>
+      <div class="hidden-xs"><%= t('.steps.build_trust') %></div>
+    </div>
+    <div class="gm-social-media-forms__step-divider gm-social-media-forms__step-divider_active"></div>
+    <div class="gm-social-media-forms__step">
```
Comment (k41n) [inline app/views/social_media_forms/new.html.erb]: It should be helper or partial obviously

Comment (k41n) [inline app/views/social_media_forms/new.html.erb]: With 3 args: image, translation key, activity (and if you rename images you can go with only 2)

File: app/views/social_media_forms/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    <p><%= t('.content.p1') %></p>
+    <p><%= t('.content.p2') %></p>
+    <div class="gm-social-media-forms__progress">
+      <div class="gm-social-media-forms__progress-blocks">
+        <%- 7.times do |i| %>
+          <div class="gm-social-media-forms__progress-block <%= 'gm-social-media-forms__progress-block_active' if i == 0 %>"></div>
+        <%- end %>
+      </div>
+      <div class="gm-social-media-forms__progress-labels">
+        <div class="gm-social-media-forms__progress-label gm-social-media-forms__progress-label_active">
```
Comment (k41n) [inline app/views/social_media_forms/new.html.erb]: Helper

File: app/views/social_media_forms/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    </div>
+  </div>
+  <div class="gm-social-media-forms__content">
+    <div class="gm-social-media-forms__content-title"><%= t('.content.title') %></div>
+    <p><%= t('.content.p1') %></p>
+    <p><%= t('.content.p2') %></p>
+    <div class="gm-social-media-forms__progress">
+      <div class="gm-social-media-forms__progress-blocks">
+        <%- 7.times do |i| %>
+          <div class="gm-social-media-forms__progress-block <%= 'gm-social-media-forms__progress-block_active' if i == 0 %>"></div>
```
Comment (k41n) [inline app/views/social_media_forms/new.html.erb]: Helper, for readability sake

File: app/views/social_media_forms/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        <div class="gm-social-media-forms__progress-label"><%= t('.content.progress.co_worker') %></div>
+        <div class="gm-social-media-forms__progress-label"><%= t('.content.progress.best_friend') %></div>
+        <div class="gm-social-media-forms__progress-label"><%= t('.content.progress.like_family') %></div>
+      </div>
+    </div>
+  </div>
+
+  <div class="gm-social-media-forms__social-networks">
+    <%- %i(linkedin facebook twitter paypal google).each do |sn_name| %>
+      <div class="gm-social-media-forms-social-network">
```
Comment (k41n) [inline app/views/social_media_forms/new.html.erb]: Partial

File: app/assets/stylesheets/social_media_forms/new.scss (hunk 5 above)
Comment (k41n) [inline app/assets/stylesheets/social_media_forms/new.scss]: In fact, thinking about 194 LoC CSS makes me sick

File: app/assets/stylesheets/social_media_forms/new.scss (hunk 4 above)
Comment (buzjuka) [inline app/assets/stylesheets/social_media_forms/new.scss]: Is it better to follow the official BEM documentation?
https://en.bem.info/methodology/quick-start/#modifier

File: app/views/social_media_forms/new.html.erb (hunk 6 above)
Comment (buzjuka) [inline app/views/social_media_forms/new.html.erb]: I myself suffer enough from this, but there is an opinion that ERB is easier to translate into JSX

**Comment by EmileKrm** [comment]
@k41n thank you for reviewing the code. Is possible to add comments to the complex parts to guide readers through the code?

File: app/assets/stylesheets/social_media_forms/new.scss (hunk 4 above)
Comment (k41n) [inline app/assets/stylesheets/social_media_forms/new.scss]: BEM is nothing more than documentation, so if you don't follow it, it's not BEM

**Pull request merged**